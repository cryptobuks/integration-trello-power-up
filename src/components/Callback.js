import React from 'react'
import './Callback.css'

export default class Callback extends React.Component {
  componentDidMount () {
    let token = window.location.hash.substring(7)
    if (window.opener) {
      window.opener.authorize(token)
    } else {
      window.localStorage.setItem('token', token)
    }
    setTimeout(function () {
      window.close()
    }, 1000)
  }

  render () {
    return (
      <div className='Callback'>
        <h1>You're set!</h1>
        <p>
          This window should close automatically. If it doesn't, go ahead and
          close it.
        </p>
      </div>
    )
  }
}
