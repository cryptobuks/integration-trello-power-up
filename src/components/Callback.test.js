import 'jsdom-global/register'
import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Callback from './Callback'
import Promise from 'bluebird'

jest.mock('../util/window.js')
jest.mock('../util/trello.js')
jest.mock('../util/config.js')
jest.mock('../util/auth.js')
jest.mock('../util/cover.js')
jest.mock('../util/attachment.js')
jest.useFakeTimers()

const setup = () => {
  const ctx = {}
  ctx.wrapper = mount(<Callback />, {
    disableLifecycleMethods: false
  })

  ctx.component = ctx.wrapper.instance()
  expect(ctx.component).toBeDefined()
  expect(toJson(ctx.wrapper)).toMatchSnapshot()
  return ctx
}

describe('Callback', () => {
  it('should render and set token to local storage', () => {
    setTimeout.mockReset()
    setup()
    jest.runAllTimers()
    Promise.attempt(() => {
      expect(window.localStorage.setItem).toBeCalledWith('token', 'some_token')
      expect(setTimeout.mock.calls.length).toBe(1) // first one is from promise
      expect(setTimeout.mock.calls[0][1]).toBe(1000)
    })
  })
  it('should render and set token opener', () => {
    setTimeout.mockReset()
    setup()
    jest.runAllTimers()

    Promise.attempt(() => {
      expect(window.opener.authorize).toBeCalledWith('some_token')
      expect(setTimeout.mock.calls.length).toBe(1) // first one is from promise
      expect(setTimeout.mock.calls[0][1]).toBe(1000)
    })
  })
})
