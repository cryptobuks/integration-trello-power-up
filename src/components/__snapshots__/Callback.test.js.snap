// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callback should render and set token opener 1`] = `
<Callback>
  <div
    className="Callback"
  >
    <h1>
      You're set!
    </h1>
    <p>
      This window should close automatically. If it doesn't, go ahead and close it.
    </p>
  </div>
</Callback>
`;

exports[`Callback should render and set token to local storage 1`] = `
<Callback>
  <div
    className="Callback"
  >
    <h1>
      You're set!
    </h1>
    <p>
      This window should close automatically. If it doesn't, go ahead and close it.
    </p>
  </div>
</Callback>
`;
